### PR TITLE
Selenium:  Added option to register java selenium 'scripts'

### PR DESCRIPTION
**Pull request by psiinon**
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: addOns/selenium/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+### Added
+- Option to register selenium 'scripts' implemented in java
```
Comment (thc202) [inline addOns/selenium/CHANGELOG.md:9]: e.g. `- Option to register selenium 'scripts' programmatically.`

File: addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/ExtensionSelenium.java [hunk 2]
```diff
@@ -128,6 +132,8 @@ public class ExtensionSelenium extends ExtensionAdaptor {
     private List<WeakReference<ProvidedBrowsersComboBoxModel>> providedBrowserComboBoxModels =
             new ArrayList<>();
 
+    private List<SeleniumScript> javaSeleniumScripts = new ArrayList<>();
```
Comment (thc202) [inline addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/ExtensionSelenium.java:135]: `customSeleniumScripts`? I'd replace "java" with "custom" everywhere. We could also introduce a new interface, that extends the existing one, and give it a better name.

File: addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/ExtensionSelenium.java [hunk 3]
```diff
@@ -1231,4 +1245,22 @@ public static boolean isConfigured(Browser browser) {
                 return true;
         }
     }
+
+    /**
+     * Register a Selenium script implemented in java.
+     *
+     * @param script the script to register
+     */
```
Comment (thc202) [inline addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/ExtensionSelenium.java:1253]: Does it worth mentioning that they are always executed synchronously?

**Comment by psiinon** [comment]
Updated as suggested except for the new interface to extend `SeleniumScript` - I think its still a reasonable class to use directly, but if not - what name do you suggest?

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**